src/poetry/console/commands/config.py

```diff
@@ -158,20 +159,19 @@ def handle(self) -> Optional[int]:
 
                 self.line(str(value))
             else:
-                values = self.unique_config_values
-                if setting_key not in values:
+                if setting_key not in self.unique_config_values:
                     raise ValueError(f"There is no {setting_key} setting.")
 
                 value = config.get(setting_key)
 
                 if not isinstance(value, str):
-                    value = json.dumps(value)
-
-                self.line(value)
+                    self.line(json.dumps(value))
```

Comment:
This logic is a bit jumbled, and I feel this change makes it moderately worse. Can we express this idiom better?

Reply:
reverted